inst/include/sampler.h

```diff
@@ -441,6 +515,8 @@ namespace smc {
         }
         else {
             nResampled = 0;
+            uRSIndices = arma::zeros<arma::Col<unsigned int> >(static_cast<int>(N));

```

Comment:
I don't use armadillo enough, I suspect, but is there a reason to first create and initialise to zero and _then_ to fill with specified values? It naively seems a touch redundant.

Reply:
Yes, looking at 518 and 519 I think only 519 ("linspace") is needed.

Also, Arma has `typedef` for the common type so instead of arma::icovec and arma::ivec are the same as a arma::Col<unsiged int>.  I guess it takes use to make it more legible as it is new -- arma::Col<unsigned int> is understood by everybody but a mouthful.

Reply:
1. Got rid of the unnecassry line.

2. On typedefs: agreed @eddelbuettel  , but I followed the existing code base like here:
* https://github.com/rcppsmc/rcppsmc/blob/6bd4ceaa976d9285cb1d867515bb003e0f3d7361/inst/include/sampler.h#L251-L255
* https://github.com/rcppsmc/rcppsmc/blob/6bd4ceaa976d9285cb1d867515bb003e0f3d7361/inst/include/sampler.h#L796

Maybe we should invest a bit of time and do a PR for re-naming everything to pure typedefs?

Reply:
Maybe.  We should pick a consistent style. I can see reasons for being explicit (everybody understands) and for being shorter (but maybe that is only used for more Armadillo-heavy projects?)

Reply:
No strong views either way; but consistent is good.